### PR TITLE
Documentation template for load balancer Monitor

### PR DESCRIPTION
**Pull request by Pushpalatha555**
---
layout: ""
page_title: "hpegl_vmaas_load_balancer_monitor Resource - vmaas-terraform-resources"
subcategory: "vmaas"
description: |-
    Loadbalancer Monitor resource facilitates creating, updating
          and deleting NSX-T Network Load Balancer Monitors.
---

# Resource hpegl_vmaas_load_balancer_monitor

-> Compatible version >= 5.4.6

Loadbalancer Monitor resource facilitates creating, updating
		and deleting NSX-T Network Load Balancer Monitors.
`hpegl_vmaas_load_balancer_monitor` resource supports NSX-T Load balancer Monitor creation.

For creating an NSX-T Load balancer Monitor, use the following examples.

## Example usage for creating NSX-T Load balancer Monitor for HTTP with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# HTTP Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_HTTP_MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_HTTP_MONITOR"
  description = "HTTP_MONITOR creating using tf"
  type        = "LBHttpMonitorProfile"
  http_monitor {
    fall_count            = 8
    interval              = 10
    monitor_port          = 50
    rise_count            = 5
    timeout               = 30
    request_body          = "request input body data"
    request_method        = "GET"
    request_url           = "https://request.com"
    request_version       = "HTTP_VERSION_1_0"
    response_data         = "Failed"
    response_status_codes = "500"
  }
}
```

## Example usage for creating NSX-T Load balancer Monitor for HTTPS with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# HTTPS Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_HTTPS_MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_HTTPS_MONITOR"
  description = "HTTPS_MONITOR creating using tf"
  type        = "LBHttpsMonitorProfile"
  https_monitor {
    fall_count            = 3
    interval              = 5
    monitor_port          = 80
    rise_count            = 3
    timeout               = 15
    request_body          = "request input body data"
    request_method        = "GET"
    request_url           = "https://test.com"
    request_version       = "HTTP_VERSION_1_1"
    response_data         = "success"
    response_status_codes = "201,200"
  }
}
```

## Example usage for creating NSX-T Load balancer Monitor for ICMP with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# ICMP Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_ICMP_MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_ICMP_MONITOR"
  description = "ICMP_MONITOR update using tf"
  type        = "LBIcmpMonitorProfile"
  icmp_monitor {
    fall_count   = 30
    interval     = 50
    monitor_port = 8
    rise_count   = 3
    timeout      = 15
    data_length  = 32
  }
}
```

## Example usage for creating NSX-T Load balancer Monitor for Passive with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# PASSIVE Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_PASSIVE_MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_PASSIVE_MONITOR"
  description = "PASSIVE_MONITOR create using tf"
  type        = "LBPassiveMonitorProfile"
  passive_monitor {
    timeout  = 15
    max_fail = 5
  }
}
```

## Example usage for creating NSX-T Load balancer Monitor for TCP with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# TCP Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_TCP_MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_TCP_MONITOR"
  description = "TCP_MONITOR create using tf"
  type        = "LBTcpMonitorProfile"
  tcp_monitor {
    fall_count    = 3
    interval      = 5
    monitor_port  = 80
    rise_count    = 3
    timeout       = 15
    request_body  = "request body data"
    response_data = "success"
  }
}
```

## Example usage for creating NSX-T Load balancer Monitor for UDP with all possible attributes

```terraform
# (C) Copyright 2022 Hewlett Packard Enterprise Development LP

# UDP Monitor
resource "hpegl_vmaas_load_balancer_monitor" "tf_UDP-MONITOR" {
  lb_id       = data.hpegl_vmaas_load_balancer.tf_lb.id
  name        = "tf_UDP-MONITOR"
  description = "UDP_MONITOR create using tf"
  type        = "LBUdpMonitorProfile"
  udp_monitor {
    fall_count    = 3
    interval      = 5
    monitor_port  = 80
    rise_count    = 3
    timeout       = 15
    request_body  = "request body data"
    response_data = "success"
  }
}
```

### Required

- `lb_id` (Number) Parent lb ID, lb_id can be obtained by using LB datasource/resource.
- `name` (String) Network loadbalancer Monitor name
- `type` (String) Provide the Supported values for monitor Type

### Optional

- `description` (String) Creating the Network Load balancer Monitor.
- `http_monitor` (Block List, Max: 1) HTTP Monitor configuration (see [below for nested schema](#nestedblock--http_monitor))
- `https_monitor` (Block List, Max: 1) Https Monitor configuration (see [below for nested schema](#nestedblock--https_monitor))
- `icmp_monitor` (Block List, Max: 1) Icmp Monitor configuration (see [below for nested schema](#nestedblock--icmp_monitor))
- `passive_monitor` (Block List, Max: 1) Passive Monitor configuration (see [below for nested schema](#nestedblock--passive_monitor))
- `tcp_monitor` (Block List, Max: 1) Tcp Monitor configuration (see [below for nested schema](#nestedblock--tcp_monitor))
- `udp_monitor` (Block List, Max: 1) Udp Monitor configuration (see [below for nested schema](#nestedblock--udp_monitor))

### Read-Only

- `id` (String) The ID of this resource.

<a id="nestedblock--http_monitor"></a>
### Nested Schema for `http_monitor`

Optional:

- `fall_count` (Number) Number of consecutive checks that must fail before marking it down.
- `interval` (Number) Set the number of times the server is tested before it is considered as DOWN
- `monitor_port` (Number) Set the value of the monitoring port.
- `request_body` (String) Enter the request body. Valid for the POST and PUT methods
- `request_method` (String) Select the method to detect the server status
- `request_url` (String) Enter the request URI for the method
- `request_version` (String) HTTP request version. Valid values are HTTP_VERSION_1_0 and HTTP_VERSION_1_1
- `response_data` (String) If the HTTP response body string and the HTTP health check response body match,then the server is considered as healthy
- `response_status_codes` (String) Enter the string that the monitor expects to match in the status line of HTTP response body.The response code is a comma-separated list
- `rise_count` (Number) Number of consecutive checks that must pass before marking it up
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

<a id="nestedblock--https_monitor"></a>
### Nested Schema for `https_monitor`

Optional:

- `fall_count` (Number) Number of consecutive checks that must fail before marking it down.
- `interval` (Number) Set the number of times the server is tested before it is considered as DOWN
- `monitor_port` (Number) Set the value of the monitoring port
- `request_body` (String) Enter the request body. Valid for the POST and PUT methods
- `request_method` (String) Select the method to detect the server status
- `request_url` (String) Enter the request URI for the method
- `request_version` (String) HTTP request version. Valid values are HTTP_VERSION_1_0 and HTTP_VERSION_1_1
- `response_data` (String) If the HTTP response body string and the HTTP health check response body match,then the server is considered as healthy
- `response_status_codes` (String) Enter the string that the monitor expects to match in the status line of HTTP response body.The response code is a comma-separated list
- `rise_count` (Number) Number of consecutive checks that must pass before marking it up
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

<a id="nestedblock--icmp_monitor"></a>
### Nested Schema for `icmp_monitor`

Optional:

- `data_length` (Number) Maximum size of the ICMP data packet
- `fall_count` (Number) Number of consecutive checks that must fail before marking it down
- `interval` (Number) Set the number of times the server is tested before it is considered as DOWN
- `monitor_port` (Number) Set the value of the monitoring port.
- `rise_count` (Number) Number of consecutive checks that must pass before marking it up
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

<a id="nestedblock--passive_monitor"></a>
### Nested Schema for `passive_monitor`

Optional:

- `max_fail` (Number) Set a value when the consecutive failures reach this value,the server is considered temporarily unavailable
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

<a id="nestedblock--tcp_monitor"></a>
### Nested Schema for `tcp_monitor`

Optional:

- `fall_count` (Number) Number of consecutive checks that must fail before marking it down.
- `interval` (Number) Set the number of times the server is tested before it is considered as DOWN
- `monitor_port` (Number) Set the value of the monitoring port.
- `request_body` (String) Enter the request body. Valid for the POST and PUT methods
- `response_data` (String) If the HTTP response body string and the HTTP health check response body matchthen the server is considered as healthy
- `rise_count` (Number) Number of consecutive checks that must pass before marking it up
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

<a id="nestedblock--udp_monitor"></a>
### Nested Schema for `udp_monitor`

Optional:

- `fall_count` (Number) Number of consecutive checks that must fail before marking it down.
- `interval` (Number) Set the number of times the server is tested before it is considered as DOWN
- `monitor_port` (Number) Set the value of the monitoring port.
- `request_body` (String) Enter the request body. Valid for the POST and PUT methods
- `response_data` (String) If the HTTP response body string and the HTTP health check response body match,then the server is considered as healthy
- `rise_count` (Number) Number of consecutive checks that must pass before marking it up
- `timeout` (Number) Set the number of times the server is tested before it is considered as DOWN

### CODE REVIEW

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# Resource hpegl_vmaas_load_balancer_monitor
+
+-> Compatible version >= 5.4.6
+
+{{ .Description | trimspace }}
+`hpegl_vmaas_load_balancer_monitor` resource supports NSX-T Load balancer Monitor creation.
+
+For creating an NSX-T Load balancer Monitor, use the following examples.
+
+-> NSX-T Monitor having HTTP,HTTPS,ICMP,Passive,TCP,UDP  considered as different monitor types. You can create either HTTP,HTTPS,ICMP,Passive,TCP,UDP, but not everything at the same time.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:18]: We don't need this. as this is handled in error

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+-> Load Balancer ID Data Source `hpegl_vmaas_load_balancer` is supported for Creating Monitor.
+
+
+## Example usage for creating NSX-T Load balancer Monitor for HTTP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for HTTP creation, attribute monitorType value `LBHttpMonitorProfile` is applicable only HTTP creation.
+
+-> Once NSX-T Load balancer Monitor for HTTP created with `LBHttpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:27]: i think we dont need these notes.

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_http_monitor.tf"}}
+
+
+## Example usage for creating NSX-T Load balancer Monitor for HTTPS with all possible attributes
+
+-> For NSX-T Load balancer Monitor for HTTPS creation, attribute monitorType value `LBHttpsMonitorProfile` is applicable only HTTPS creation.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:34]: ditto, may not need them. Sync with Jinu once.

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+-> Once NSX-T Load balancer Monitor for HTTPs created with `LBHttpsMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_https_monitor.tf"}}
+
+
+## Example usage for creating NSX-T Load balancer Monitor for ICMP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for ICMP creation, attribute monitorType value `LBIcmpMonitorProfile` is applicable only ICMP creation.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:43]: not needed.

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+-> Once NSX-T Load balancer Monitor for ICMP created with `LBIcmpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_icmp_monitor.tf"}}
+
+
+## Example usage for creating NSX-T Load balancer Monitor for Passive with all possible attributes
+
+-> For NSX-T Load balancer Monitor for Passive creation, attribute monitorType value `LBPassiveMonitorProfile` is applicable only Passive creation.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:52]: not needed, remove it

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+-> For NSX-T Load balancer Monitor for TCP creation, attribute monitorType value `LBTcpMonitorProfile` is applicable only TCP creation.
+
+-> Once NSX-T Load balancer Monitor for TCP created with `LBTcpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_tcp_monitor.tf"}}
+
+
+## Example usage for creating NSX-T Load balancer Monitor for UDP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for UDP creation, attribute monitorType value `LBUdpMonitorProfile` is applicable only UDP creation.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:70]: not needed, remove it

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 2 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:27]: done

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 1 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:18]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 3 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:34]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 4 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:43]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 5 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:52]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+-> For NSX-T Load balancer Monitor for Passive creation, attribute monitorType value `LBPassiveMonitorProfile` is applicable only Passive creation.
+
+-> Once NSX-T Load balancer Monitor for Passive created with `LBPassiveMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_passive_monitor.tf"}}
+
+
+## Example usage for creating NSX-T Load balancer Monitor for TCP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for TCP creation, attribute monitorType value `LBTcpMonitorProfile` is applicable only TCP creation.
```
[the remark this answers is not in the document]
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:61]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+## Example usage for creating NSX-T Load balancer Monitor for HTTP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for HTTP creation, attribute monitorType value `LBHttpMonitorProfile` is applicable only HTTP creation.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_http_monitor.tf"}}
+
+## Example usage for creating NSX-T Load balancer Monitor for HTTPS with all possible attributes
+
+-> Once NSX-T Load balancer Monitor for HTTPs created with `LBHttpsMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:26]: not needed.

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_https_monitor.tf"}}
+
+## Example usage for creating NSX-T Load balancer Monitor for ICMP with all possible attributes
+
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:31]: not needed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+-> Once NSX-T Load balancer Monitor for ICMP created with `LBIcmpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_icmp_monitor.tf"}}
+
+## Example usage for creating NSX-T Load balancer Monitor for Passive with all possible attributes
+
+-> Once NSX-T Load balancer Monitor for Passive created with `LBPassiveMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:38]: not needed, please remove these in other places as well

Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:38]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 9 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:31]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 8 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl:26]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_passive_monitor.tf"}}
+
+## Example usage for creating NSX-T Load balancer Monitor for TCP with all possible attributes
+
+-> Once NSX-T Load balancer Monitor for TCP created with `LBTcpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:44]: not needed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+{{tffile "examples/resources/hpegl_vmaas_load_balancer_monitor/nsx_t_lb_tcp_monitor.tf"}}
+
+## Example usage for creating NSX-T Load balancer Monitor for UDP with all possible attributes
+
+-> Once NSX-T Load balancer Monitor for UDP created with `LBUdpMonitorProfile`, we can not modify the attribute during the update operation and its disabled from UI.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:50]: not needed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+-> Compatible version >= 5.4.6
+
+{{ .Description | trimspace }}
+`hpegl_vmaas_load_balancer_monitor` resource supports NSX-T Load balancer Monitor creation.
+
+For creating an NSX-T Load balancer Monitor, use the following examples.
+
+## Example usage for creating NSX-T Load balancer Monitor for HTTP with all possible attributes
+
+-> For NSX-T Load balancer Monitor for HTTP creation, attribute monitorType value `LBHttpMonitorProfile` is applicable only HTTP creation.
```
Comment (manjunath-batakurki) [inline templates/resources/vmaas_load_balancer_monitor.md.tmpl:20]: not needed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 11 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 12 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl]: removed

File: templates/resources/vmaas_load_balancer_monitor.md.tmpl (hunk 13 above)
Comment (Pushpalatha555) [reply templates/resources/vmaas_load_balancer_monitor.md.tmpl]: removed

File: internal/resources/resource_lb_monitors.go [hunk 14]
```diff
@@ -58,8 +58,8 @@ func LoadBalancerMonitor() *schema.Resource {
 		CreateContext: loadbalancerMonitorCreateContext,
 		DeleteContext: loadbalancerMonitorDeleteContext,
 		CustomizeDiff: monitorCustomDiff,
-		Description: `loadbalancer Monitor resource facilitates creating,updating
-		and deleting NSX-T Network Load Balancers.`,
+		Description: `Loadbalancer Monitor resource facilitates creating,updating
```
[the remark this answers is not in the document]
Comment (Pushpalatha555) [reply internal/resources/resource_lb_monitors.go]: updated

**Review by ravikumar29524 — APPROVED** [review]

**Review by manjunath-batakurki — APPROVED** [review]

**Pull request merged**